data_miner_service/__init__.py

```diff
@@ -117,6 +127,18 @@ def extract_domains(data):
                 pass
     return final_domains
 
+# hack of a parser to extract potential URLs (Links) from data
+def extract_urls(data):
+    pattern = r'(http|ftp|https)://([\w_-]+(?:(?:\.[\w_-]+)+))([\w.,@?^=%&:/~+#-]*[\w@?^=%&/~+#-])?'
+    results = re.findall(pattern,data)
+    urls = [each for each in results if len(each) >0]
+    final_urls = []
+    for item in urls:
+        url = item[0]+"://"+item[1]+item[2]
```

Comment:
You could simplify this code by adding one more regex group and taking the first entry. For example:

  pattern = r'((http|ftp|https)://([\w_-]+(?:(?:\.[\w_-]+)+))([\w.,@?^=%&:/~+#-]*[\w@?^=%&/~+#-])?)'
  results = re.findall(pattern,data)
  final_urls = [result[0] for result in results